tests/Smoke/CiIntegrationTest.php

```diff
@@ -136,20 +136,23 @@ public function testIntegration(
 If you need help while solving warnings, ask at https://gitter.im/PHP-CS-Fixer, we will help you!
 ';
 
-        $executionDetails = "Loaded config default from \".php_cs.dist\".
-${expectedResult3Files}
-Legend: ?-unknown, I-invalid file syntax, file ignored, S-Skipped, .-no changes, F-fixed, E-error";
-
-        $this->assertRegExp(
-            sprintf(
-                '/^(%s)?(%s)?%s$/',
-                preg_quote($optionalIncompatibilityWarning, '/'),
-                preg_quote($optionalXdebugWarning, '/'),
-                preg_quote($executionDetails, '/')
-            ),
-            $result3->getError()
+        $pattern = sprintf(
+            '/^(?:%s)?(?:%s)?%s\n([\.S]{%d})\n%s$/',
+            preg_quote($optionalIncompatibilityWarning, '/'),
+            preg_quote($optionalXdebugWarning, '/'),
+            preg_quote('Loaded config default from ".php_cs.dist".', '/'),
+            strlen($expectedResult3Files),
+            preg_quote('Legend: ?-unknown, I-invalid file syntax, file ignored, S-Skipped, .-no changes, F-fixed, E-error', '/')
         );
 
+        $this->assertRegExp($pattern, $result3->getError());
+
+        preg_match($pattern, $result3->getError(), $matches);
+
+        $this->assertArrayHasKey(1, $matches);
+        $this->assertSame(substr_count($expectedResult3Files, '.'), substr_count($matches[1], '.'));
+        $this->assertSame(substr_count($expectedResult3Files, 'S'), substr_count($matches[1], 'S'));
```

Comment:
What if one would add case with `'F'` ? he wouldn't know that `F` wasn't tested and only `'.'` and `'S'` were.
Would be nice to reuse `\PhpCsFixer\Console\Output\ProcessOutput::$eventStatusMap` (after little refactoring, don't use private property in tests ;) ).

alternative - ensure that `(count of '.' + count of 'S') === strlen($expectedResult3Files)`

Reply:
I have no concept for "little refactoring" (best I came with is extracting statuses to classes value object `Status` and collection `StatusCollection` - it's not a little change and seems like overkill - we won't be extending the list anyway) and doing it only for the use of this test does not feel right either.

I've ensured in RegEx that only `.` and `S` are in the list, so if someone add case with `F` then `assertRegExp` will fail - handling all 6 type seemed a little YAGNI, or should I go in that direction?

Reply:
nah, if you ensured only `.S` are allowed, it's fine as well